Content.Shared/DeltaV/CartridgeLoader/Cartridges/StockTradingUiState.cs

```diff
@@ -0,0 +1,59 @@
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.CartridgeLoader.Cartridges;
+
+[Serializable, NetSerializable]
+public sealed class StockTradingUiState(
+    List<StockCompanyStruct> entries,
+    Dictionary<string, int> ownedStocks,
+    float balance)
+    : BoundUserInterfaceState
+{
+    public readonly List<StockCompanyStruct> Entries = entries;
+    public readonly Dictionary<string, int> OwnedStocks = ownedStocks;
+    public readonly float Balance = balance;
+}
+
+// No structure, zero fucks given
+[DataDefinition, Serializable]
+public partial struct StockCompanyStruct
+{
+    /// <summary>
+    /// The internal name/key of the company. Should not contain spaces or special characters.
+    /// </summary>
+    [DataField(required: true)]
+    public string Name;
```

Comment:
why does name need to exist, have it be a list of companies and have network messages use int index

nothing prevents you from having 2 companies with the same display name and different name, and you can easily know the index from position in the UI anyway

Reply:
i realized way too late that messages rely on names and so they can't have spaces or special characters so displayname was an easy workaround without reworking the entire thing, but yeah that does make more sense

Reply:
yes rework the entire thing :trollface: